azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java

```diff
@@ -16,39 +16,275 @@
+import org.jclouds.azurecompute.domain.NetworkSecurityGroup;
 import org.jclouds.azurecompute.domain.OSImage;
+import org.jclouds.azurecompute.domain.Operation;
 import org.jclouds.azurecompute.domain.RoleSize;
+import org.jclouds.azurecompute.domain.StorageService;
+import org.jclouds.azurecompute.domain.StorageServiceParams;
+import org.jclouds.azurecompute.options.AzureComputeTemplateOptions;
 import org.jclouds.compute.ComputeServiceAdapter;
+import org.jclouds.compute.domain.OsFamily;
 import org.jclouds.compute.domain.Template;
+import org.jclouds.compute.reference.ComputeServiceConstants;
+import org.jclouds.domain.LoginCredentials;
+import org.jclouds.logging.Logger;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Sets;
 
 /**
  * defines the connection between the {@link AzureComputeApi} implementation and the
  * jclouds {@link org.jclouds.compute.ComputeService}
  */
 @Singleton
-public class AzureComputeServiceAdapter implements ComputeServiceAdapter<Deployment, RoleSize, OSImage, String> {
+public class AzureComputeServiceAdapter implements ComputeServiceAdapter<Deployment, RoleSize, OSImage, Location> {
+
+   private static final String DEFAULT_VIRTUAL_NETWORK_NAME = "jclouds-virtual-network";
+   private static final String DEFAULT_ADDRESS_SPACE_ADDRESS_PREFIX = "10.0.0.0/20";
+   private static final String DEFAULT_SUBNET_NAME = "jclouds-1";
+   private static final String DEFAULT_SUBNET_ADDRESS_PREFIX = "10.0.0.0/23";
+   private static final String DEFAULT_STORAGE_SERVICE_TYPE = "Standard_GRS";
+
+   private static final String DEFAULT_LOGIN_USER = "jclouds";
+   private static final String DEFAULT_LOGIN_PASSWORD = "Azur3Compute!";
+   private static final String DEFAULT_STORAGE_ACCOUNT_PREFIX = "jclouds";
+
+   @Resource
+   @Named(ComputeServiceConstants.COMPUTE_LOGGER)
+   protected Logger logger = Logger.NULL;
+
+   private final AzureComputeApi api;
+   private final String subscriptionId;
+   private final long operationTimeout;
+   private Predicate<String> operationSucceeded;
+
+   @Inject
+   public AzureComputeServiceAdapter(final AzureComputeApi api, @Named(SUBSCRIPTION_ID) String subscriptionId, @Named(OPERATION_TIMEOUT) long operationTimeout) {
+      this.api = checkNotNull(api, "api");
+      this.subscriptionId = checkNotNull(subscriptionId, "subscriptionId");
+      this.operationTimeout = checkNotNull(operationTimeout, "operationTimeout");
+      this.operationSucceeded = retry(new Predicate<String>() {
+         public boolean apply(String input) {
+            return api.getOperationApi().get(input).status() == Operation.Status.SUCCEEDED;
+         }
+      }, operationTimeout, 5, 5, SECONDS);
```

Comment:
Consider moving this predicate to a provider method in the Guice configuration module so it can be injected and reused (reused in live tests too?).